_shared/local_extensions.py

```diff
@@ -110,11 +113,84 @@ def include_exists(self, context, path):
+
+        Return the contents of the project's .cookiecutter/includes/{path} JSON
+        file deserialized into a Python object.
+
+        Return None if the project has no .cookiecutter/includes/{path} file.
+        """
+        return json.load(self._open(context, path))
+
+    @pass_context
+    def supervisor_settings(self, context):
+        """Return the settings from the project's supervisor.json file.
+
+        Or return the default supervisor settings.
+        """
+        # Are we rendering the production supervisord.conf or the development environment's supervisor-dev.conf?
+        if context.name.endswith("supervisord-dev.conf"):
+            env = "dev"
+        else:
+            assert context.name.endswith("supervisord.conf")
+            env = "prod"
+
+        try:
+            settings = self.include_json(context, "supervisor.json")
+        except FileNotFoundError:
+            settings = {}
+
+        if not settings.get("programs"):
+            if context["cookiecutter"]["_directory"] == "pyramid-app":
+                settings["programs"] = {"web": {"settings": {}}}
+                if env == "dev":
+                    settings["programs"]["web"]["settings"][
+                        "command"
+                    ] = f"gunicorn --bind :{context['cookiecutter']['port']} --workers 1 --reload --timeout 0 --paste conf/development.ini"
+                else:
+                    assert env == "prod"
+                    settings["programs"]["web"]["settings"][
+                        "command"
+                    ] = f"newrelic-admin run-program gunicorn --paste conf/production.ini --bind 0.0.0.0:{context['cookiecutter']['port']}"
+            else:
+                settings["programs"] = {
+                    "app": {
+                        "settings": {
+                            "command": f"python3 -m {context['cookiecutter']['package_name']}",
+                        }
+                    }
+                }
+
+        for program in settings["programs"].values():
+            program.setdefault("dev", True)
+            program.setdefault("prod", True)
+            program.setdefault("settings", {})
+            program["settings"].setdefault("stdout_events_enabled", "true")
+            program["settings"].setdefault("stderr_events_enabled", "true")
+            program["settings"].setdefault("stopsignal", "KILL")
+            program["settings"].setdefault("stopasgroup", "true")
+
+        if context.name.endswith("supervisord.conf"):
+            test = lambda program_settings: program_settings["prod"]
+        else:
+            assert context.name.endswith("supervisord-dev.conf")
+            test = lambda program_settings: program_settings["dev"]
+
+        settings["programs"] = {
+            program_name: program_settings
+            for program_name, program_settings in settings["programs"].items()
+            if program_settings[env]
+        }
+
+        return settings
```

Comment:
This Supervisor-specific method is implemented in Python partially because it's called by two templates (`supervisord.conf` and `supervisord-dev.conf`) and with cookiecutter there's no way to share template or macro code between files, but this global Python function is available to all templates.

And partially because the logic in this function (all the setting defaults and stuff) is really really hard to implement in Jinja2.